server/conditions/conditions.go

```diff
@@ -0,0 +1,109 @@
+package conditions
+
+import (
+	"github.com/bitdecaygames/fireport/server/actions"
+	"github.com/bitdecaygames/fireport/server/cards"
+	"github.com/bitdecaygames/fireport/server/pogo"
+	"sort"
+)
+
+// Condition checks each ActionGroup for a specific condition and modifies that ActionGroup if necessary
+type Condition interface {
+	Apply(gameState *pogo.GameState, actionGroup []actions.Action) error
+}
+
+// ProcessConditions with a GameState, Inputs, and Conditions, generate the necessary and valid list of actions to get to the next state
+func ProcessConditions(currentState *pogo.GameState, inputs []pogo.GameInputMsg, conditions []Condition) (*pogo.GameState, error) {
```

Comment:
This is probably the craziest method I've ever written.  Mostly because of golang.  Although, there is a good chance I could be doing something in a really ugly way because I'm not a golang guy.

This is where the majority of the stuff we talked about at the beginning of the week is happening.  The list of inputs from the clients is broken down, grouped, split up, grouped again, and then partitioned into what I referred to in #39 as `ActionGroup`s.  I'm utterly amazed that this function works at all.  But the tests in conditions_test.go prove it is doing the job.